test/unit/org/apache/cassandra/tools/StandaloneSplitterTest.java

```diff
@@ -52,5 +67,101 @@ public void testStandaloneSplitter_NoArgs()
+            assertThat(tool.getCleanedStderr(), CoreMatchers.containsStringIgnoringCase("No valid sstables to split"));
+            assertEquals(1,tool.getExitCode());
+        }
+        assertCorrectEnvPostTest();
+    }
+
+    @Test
+    public void testFlagArgs()
+    {
+        Arrays.asList("--debug", "--no-snapshot").forEach(arg -> {
+            try (ToolRunner tool = runner.invokeClassAsTool(StandaloneSplitter.class.getName(), arg, "mockFile"))
+            {
+                assertThat("Arg: [" + arg + "]", tool.getStdout(), CoreMatchers.containsStringIgnoringCase("Skipping inexisting file mockFile"));
+                assertThat("Arg: [" + arg + "]", tool.getCleanedStderr(), CoreMatchers.containsStringIgnoringCase("No valid sstables to split"));
+                assertEquals(1,tool.getExitCode());
+            }
+            assertCorrectEnvPostTest();
+        });
+    }
+
+    @Test
+    public void testSizeArg()
+    {
+        Arrays.asList(Pair.of("-s", ""), Pair.of("-s", "w"), Pair.of("--size", ""), Pair.of("--size", "w"))
+              .forEach(arg -> {
+                  try
+                  {
+                      runner.invokeClassAsTool(StandaloneSplitter.class.getName(),
+                                               arg.getLeft(),
+                                               arg.getRight(),
+                                               "mockFile");
+                      fail("Shouldn't be able to parse wrong input as number");
+                  }
+                  catch(RuntimeException e)
+                  {
+                      if (!(e.getCause() instanceof NumberFormatException))
+                          fail("Should have failed parsing a non-num.");
+                  }
+              });
+
+        Arrays.asList(Pair.of("-s", "0"),
+                      Pair.of("-s", "1000"),
+                      Pair.of("-s", "-1"),
+                      Pair.of("--size", "0"),
+                      Pair.of("--size", "1000"),
+                      Pair.of("--size", "-1"))
+              .forEach(arg -> {
+                  try (ToolRunner tool = runner.invokeClassAsTool(StandaloneSplitter.class.getName(),
+                                                                  arg.getLeft(),
+                                                                  arg.getRight(),
+                                                                  "mockFile"))
+                  {
+                      assertThat("Arg: [" + arg + "]", tool.getStdout(), CoreMatchers.containsStringIgnoringCase("Skipping inexisting file mockFile"));
+                      assertThat("Arg: [" + arg + "]", tool.getCleanedStderr(), CoreMatchers.containsStringIgnoringCase("No valid sstables to split"));
+                      assertEquals(1,tool.getExitCode());
+                  }
+                  assertCorrectEnvPostTest();
```

Comment:
general comment, any reason this isn't an `@After` method?  that way you don't need to add constantly?

Reply:
More tests will be added when we tackle the specific ticket to add proper testing for the specific tool. Those will be _appended_ (notice `OrderedJUnit4ClassRunner`) and depending on what the tool does/loads the post test env can be different. So `assertCorrectEnvPostTest` only applies on a case by case basis despite for args testing it applies always.